pkg/controller/packet/server/device/managed.go

```diff
@@ -171,12 +187,26 @@ func (e *external) Update(ctx context.Context, mg resource.Managed) (resource.Ex
 		return resource.ExternalUpdate{}, errors.New(errNotDevice)
 	}
 
-	_, _, err := e.client.Get(d.Status.ID)
+	actual, _, err := e.client.Get(d.Status.ID, nil)
 	if err != nil {
 		return resource.ExternalUpdate{}, errors.Wrap(err, errGetDevice)
 	}
 
-	// TODO: compare returned to desired state and update if necessary
+	req := &packngo.DeviceUpdateRequest{}
+	if d.Status.Hostname != actual.Hostname {
```

Comment:
I usually like to have a func in `clients/` that is something like `device.NeedsUpdate()` (example [here](https://github.com/crossplaneio/stack-gcp/blob/48dc96d9685094e9a1fc3f2eab321991fa46f33d/pkg/clients/cloudmemorystore/cloudmemorystore.go#L136)) just to keep the controller a little cleaner. Totally a personal preference though!

Reply:
I agree it should be in a different function. I will encapsulate this so it is neater. I just wanted to get this started.